src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -4311,6 +4311,7 @@ public sealed partial class OperatingSystem : System.ICloneable, System.Runtime.
         public static bool IsAndroid() { throw null; }
         public static bool IsAndroidVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0) { throw null; }
         public static bool IsBrowser() { throw null; }
+        public static bool IsWASI() { throw null; }
```

Comment:
This needs to go through API review

Reply:
https://github.com/dotnet/runtime/issues/78389